index.bs

```diff
@@ -1215,6 +1215,383 @@ implementation of this step is specific to the user agent.", the controlling
+
+<!--
+  TODO:
+
+  Put in something like this:
+
+streaming-capabilities-request {
+  request
+}
+
+streaming-capabilities-response {
+  response
+  capabilities: streaming-capabilities
+}
+
+streaming-capabilities = {
+  receive-audio: [* receive-audio-capability]
+  receive-video: [* receive-video-capability]
+  receive-data: [* receive-data-capability]
+}
+
+
+receive-audio-capability = {
+  codec-name: text
+  codec-parameters: [* codec-parameter]
+}
+
+receive-video-capability = {
+  codec-name: text
+  codec-parameters: [* codec-parameter]
+  ? max-resolution: video-resolution
+  ? max-frames-per-second: float
+  ? max-pixels-per-second: float
+}
+
+receive-data-capability = {
+  format: text ; we define "text-cue" for text cues
+  format-parameters: codec-parameters
+}
+
+codec-parameter = {
+  key: text
+  value: bytes
+}
+
+-->
+
+Sessions {#streaming-sessions}
+------------------------------------
+
+<!--
+  TODO: Put in something like this
+
+streaming-session-start-request = {
+  request
+  streaming-session-id: uint
+  stream-offers: [* media-stream-offer]
+  stats-interval: microseconds ; please send receiver stats this often
+}
+
+streaming-session-start-response = {
+  response
+  stream-requests: [* media-stream-request]
+  stats-interval: microseconds ; please send sender stats this often
+}
+
+streaming-session-modify-request = {
+  request
+  streaming-session-id: uint
+  stream-requests: [* media-stream-request]
+}
+
+streaming-session-modify-response = {
+  response
+}
+
+streaming-session-stop-request = {
+  request
+  streaming-session-id: uint
+}
+
+streaming-session-stop-response = {
+  response
+}
+
+streaming-session-stop-event= {
+  streaming-session-id: uint
+}
+
+media-stream-offer = {
+  media-stream-id: uint
+  ? friendly-name: text
+  ? audio: [* audio-encoding-offer]
+  ? video: [* video-encoding-offer]
+  ? data: [* data-encoding-offer]
+}
+
+media-stream-request = {
+  media-stream-id: uint
+  ? audio: [* audio-encoding-request]
+  ? video: video-encoding-request
+  ? data: [* data-encoding-request]
+}
+
+audio-encoding-offer = {
+  encoding-id: uint
+  codec-name: text
+  codec-parameters: [* codec-parameter]
+  time-scale: uint ; AKA sample rate or clock rate
+  default-duration: uint ; AKA ptime
+}
+
+video-encoding-offer = {
+  encoding-id: uint
+  codec-name: text
+  codec-parameters: [* codec-parameter]
+  time-scale: uint ; AKA clock rate; probably 90khz
+  ? default-duration: uint ; AKA ptime
+  ? default-rotation: video-rotation
+}
+
+data-encoding-offer = {
+  encoding-id: uint
+  format: text ; "text-cue" defined to mean text cues
+  format-parameters: [* codec-parameter] ; format-specific
+  time-scale: uint ; AKA clock rate; probably 1mhz
+}
+
+audio-stream-request = {
+  encoding-id: uint
+}
+
+video-stream-request = {
+  encoding-id: uint
+  ? target-resolution: video-resolution ; displayed resolution; try to be slightly above; default is unknown
+  ? max-framerate: float
```

Comment:
might be worth specifically specifying float16 / float32 / float64 here

Reply:
I'll put that in a future PR, but I'll make it float16